File: 1605-find-valid-matrix-given-row-and-column-sums/1605-find-valid-matrix-given-row-and-column-sums.py
class Solution:
    def restoreMatrix(self, rowSum: List[int], colSum: List[int]) -> List[List[int]]:
        m, n = len(rowSum), len(colSum)
        result = [[0 for i in range(n)] for j in range(m)]
        for r in range(m):
            for c in range(n):
                result[r][c] = min(rowSum[r], colSum[c])
                rowSum[r] -= result[r][c]
                colSum[c] -= result[r][c]
        return result;

File: 1605-find-valid-matrix-given-row-and-column-sums/1605-find-valid-matrix-given-row-and-column-sums.cpp
class Solution {
public:
    vector<vector<int>> restoreMatrix(vector<int>& rowSum, vector<int>& colSum) {
        int m = rowSum.size(), n = colSum.size();
        vector<vector<int>> result(m, vector<int>(n, 0));
        for (int r = 0; r < m; r++) {
            for (int c = 0; c < n; c++) {
                int mn = min(rowSum[r], colSum[c]);
                rowSum[r] -= mn; colSum[c] -= mn;
                result[r][c] = mn;
            }
        }
        return result;
    }
};